query/src/main/java/io/keen/client/java/SavedQueryPut.java

```diff
@@ -0,0 +1,111 @@
+package io.keen.client.java;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import io.keen.client.java.http.HttpMethods;
+
+/**
+ * Represents PUT requests performed against the Saved/Cached Queries API.
+ *
+ * @author masojus
+ */
+class SavedQueryPut extends SavedQueryRequest {
+    private final KeenQueryRequest query;
+    private final int refreshRate;
+
+    private final Map<String, Object> miscProperties;
+
+
+    SavedQueryPut(String queryName,
+                  String displayName,
+                  KeenQueryRequest query,
+                  int refreshRate,
+                  Map<String, ?> miscProperties) {
+        super(HttpMethods.PUT, true /* needsMasterKey */, queryName, displayName);
```

Comment:
Sometimes I like to use enums for that reason, just another option. Sometimes that's overkill though. Named parameters would be nice.